File: backend/app/models/producto.py
from sqlmodel import SQLModel, Field, Relationship
from typing import List, Optional
from datetime import datetime

class Producto(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    nombre: str = Field(max_length=150)
    descripcion: Optional[str] = None
    precio_base: float = Field(ge=0)
    stock_cantidad: int = Field(default=0, ge=0)
    disponible: bool = Field(default=True)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    deleted_at: Optional[datetime] = Field(default=None)

    categorias: List["ProductoCategoria"] = Relationship(back_populates="producto")
    ingredientes: List["ProductoIngrediente"] = Relationship(back_populates="producto")

File: frontend/src/types/producto.ts
import type { Categoria } from './categoria'
import type { Ingrediente } from './ingrediente'

export interface Producto {
  id: number
  nombre: string
  descripcion?: string
  precio_base: number
  stock_cantidad: number
  disponible: boolean
  created_at: string
  updated_at: string
}

export interface ProductoCreate {
  nombre: string
  descripcion?: string
  precio_base: number
  stock_cantidad: number
  disponible: boolean
  categoria_ids: number[]
  ingrediente_ids: number[]
}

export interface ProductoDetalle {
  id: number
  nombre: string
  descripcion?: string
  precio_base: number
  stock_cantidad: number
  disponible: boolean
  created_at: string
  updated_at: string
  categorias: Categoria[]
  ingredientes: Ingrediente[]
}